lib/queries/otu/filter.rb

```diff
@@ -135,8 +136,13 @@ def geographic_area_scope
       end
 
       gi_sql = GeographicItem.contained_by_where_sql(target_geographic_item_ids)
-      co_ids = CollectionObject.joins(:geographic_items).where(gi_sql).distinct.ids
-      ad_ids = AssertedDistribution.joins(:geographic_items).where(gi_sql).distinct.ids
+      if project_id.present?
+        co_ids = CollectionObject.with_project_id(project_id).joins(:geographic_items).where(gi_sql).distinct.ids
```

Comment:
These are the real problem with speed and scoping, we aren't using joins, but rather arrays of ids.